### ISSUE TITLE
Subscription issue with Home Assistant

### ISSUE THREAD

**Issue by HHashi** [fvanroie/MqttClientPlugin#6]
Hi,

Sorry it's me again!

So previously I was running the Mosquitto broker on my PC, where Rainmeter was also running. Things were running smoothly with that particular set up.

I wanted to move the MQTT broker onto a Raspeberry Pi (on the same local network) so that it's not reliant on my desktop being on. The Raspberry Pi is running [Home Assistant](https://www.home-assistant.io/), which has a Mosquitto plugin, so I installed that.

Mosquitto was configured to allow anonymous connections according to the Home Assistant docs, but looking at the Rainmeter logs, I could see that I was getting a _NotAuthorized_ error. I hadn't included `User` or `Password` in the `[mqttServer]` section in the .ini file:

```
[mqttServer]
Measure=Plugin
Plugin=MqttClient
Server=192.168.1.169 ; Rapsberry Pi's IP
```

After a bit of digging around, I found that for this particular set up (your plugin, Home Assistant, the Mosquitto plugin and anonymous connections), you need to comment out `.WithCredentials(username, SecureStringToString(password))`  in _/MqttClientPlugin/MqttClientMeasure.cs_ (line 342), prior to building and moving _MqttClient.dll_ to the plugins folder.

```
var options = new MqttClientOptionsBuilder()
    .WithClientId(clientID)
    .WithTcpServer(server, port)
    // .WithCredentials(username, SecureStringToString(password))
    .WithCleanSession(true)    // must be true for Managed Client, easier reconnects
    .Build();
```

If I knew C# I would have liked to have added a conditional statement to only leave out the above when the username & password aren't set, but hey ho.

Thought I'd post in case someone else comes across a similar issue. Would have saved me 3 hours :sweat_smile:

Thanks again for your work on this plugin!

**Comment by fvanroie** [comment]
Thanks for your report, I'll implement this as an enhancement in an upcoming version.
It is mostly me using this plugin, so it isn't heavily field tested as such.

I'm glad you find it helpful.

**Comment by fvanroie** [comment]
This has been fixed in v0.2.0
